libraries/AP_Math/AP_Math.cpp

```diff
@@ -128,6 +128,26 @@ float throttle_curve(float thr_mid, float alpha, float thr_in)
     return thr_out;
 }
 
+/*
+ * Convert any base number to any base number. Example octal(8) to decimal(10)
+ * baseIn: base of input number
+ * baseOut: base of output number
+ * inputNumber: value currently in base "baseIn" to be converted to base "baseOut"
+ *
+ * Example: convert ADSB squawk octal "1200" stored in memory as 0x0280 to 0x04B0
+ *          uint16_t squawk_decimal = convertMathBase(8, 10, squawk_octal);
+ */
+uint32_t convertMathBase(const uint8_t baseIn, const uint8_t baseOut, uint32_t inputNumber)
+{
+    uint32_t outputNumber = 0;
+
+    for (uint8_t i=0; inputNumber != 0; i++) {
+        outputNumber += (inputNumber % baseOut) * powf(float(baseIn), i);
```

Comment:
per devcall:
- change powf() to pow() integer maths
- move to adsb library
- hardcode for specific bases for ADSB

Reply:
Applied.